### ISSUE TITLE
Failed to decode packet 'clientbound/minecraft:recipe_book_add'

### ISSUE THREAD

**Issue by kbingham1993** [Glitchfiend/SereneSeasons#513]
### What's the issue you encountered?

It crashed when I acquired a clock in an attempt to make a calendar. Now, I am unable to log into the server without uninstalling the mod  cannot

### How can the issue be reproduced?

Have paper and a clock in inventory, it attempts to assign you the calendar recipe but is unable too and crashes you out of the server. If you have paper and a clock in inventory, it attempts to assign you the calendar recipe but is unable to and crashes.

### Logs

https://gist.github.com/kbingham1993/ae8ed050bad500772742f750f9ea9b78

### Mod Version

SereneSeasons-fabric-1.21.3-10.2.0.1

**Comment by Bret06** [comment]
I don't believe the issue is caused by the same thing it was for me. However, while playing with one of our datapacks one of my staff messaged me and sent this exact same error code with a slightly different reason for the crash:
`Caused by: java.lang.IllegalArgumentException: Invalid key. Must be [a-z0-9/._-]: `
This was caused by a recipe in our datapack that was incorrectly generated by Misode causing the server to mismatch the recipe packets. Basically in recipes attribute components in newer versions there is a `id` and `type` value. When I used misode to generate the recipe it gave me this:
```            "minecraft:attribute_modifiers": {
                "modifiers": [
                    {
                        "type": "minecraft:attack_damage",
                        "id": "minecraft:",
                        "amount": 0,
                        "operation": "add_value"
                    }
                ],
                "show_in_tooltip": false
            }
```
Typically, in Minecraft datapacks when you receive this error: ``Invalid key. Must be [a-z0-9/._-]:`` it means there is a error somewhere in a definition. It took 5 seconds of skimming the recipe json before I noticed how off the id value looked:
``"id": "minecraft:",``

All I did was copy and paste the type to the id, which was purely guess work but it WORKED FIRST TRY. So instead of this:
```            "minecraft:attribute_modifiers": {
                "modifiers": [
                    {
                        "type": "minecraft:attack_damage",
                        "id": "minecraft:",
                        "amount": 0,
                        "operation": "add_value"
                    }
                ],
                "show_in_tooltip": false
            }
```
Do this:
```             "minecraft:attribute_modifiers": {
                "modifiers": [
                    {
                        "type": "minecraft:attack_damage",
                        "id": "minecraft:attack_damage",
                        "amount": 0,
                        "operation": "add_value"
                    }
                ],
                "show_in_tooltip": false
            }
```
